src/graphql/mutations/password_reset.rs

```diff
@@ -0,0 +1,116 @@
+use async_graphql::{Context, Object};
+use sea_orm::{
+    ActiveValue, ColumnTrait, DatabaseConnection, EntityTrait, IntoActiveModel, QueryFilter,
+};
+use sendgrid::SGClient;
+use tracing::{event, Level};
+
+use crate::{
+    error::new_err,
+    schema::{self, password_reset_tokens, users},
+};
+
+#[derive(Default)]
+pub struct PasswordResetMutation;
+const RESET_URL_BASE: &str = "https://lumina.earth/reset?token=";
+#[Object(rename_fields = "snake_case", rename_args = "snake_case")]
+impl PasswordResetMutation {
+    #[graphql()]
+    pub async fn reset_password(
+        &self,
+        ctx: &Context<'_>,
+        email: String,
+    ) -> async_graphql::Result<&str> {
+        let db = ctx.data_unchecked::<DatabaseConnection>();
+        let s_g_client = ctx.data_unchecked::<SGClient>();
+
+        let user = users::Entity::find()
+            .filter(users::Column::Email.contains(&email))
+            .one(db)
+            .await?
+            .ok_or_else(|| new_err("USER_NOT_FOUND", &format!("User not found: {}", &email)))?;
+
+        let mut new_reset_token = password_reset_tokens::ActiveModel::default();
+        new_reset_token.id = ActiveValue::Set(uuid::Uuid::new_v4());
+        new_reset_token.user_id = ActiveValue::Set(user.id);
+
+        let token = match password_reset_tokens::Entity::insert(new_reset_token)
+            .exec_with_returning(db)
+            .await
+        {
+            Err(db_error) => {
+                event!(Level::ERROR, "{}", db_error);
+                return Err(new_err(
+                    "TOKEN_CREATION_ERROR",
+                    "unable to create new token",
+                ));
+            }
+
+            Ok(token) => token.id,
+        };
+        let reset_url = RESET_URL_BASE.to_owned() + &token.simple().to_string();
+        let reset_text = format!("go to {} to reset your password", reset_url);
+        let reset_password_mail = sendgrid::Mail::new()
+            .add_from("no-reply@lumina.earth")
+            .add_text(&reset_text)
+            .add_subject("Lumina: Your password reset link!")
+            .add_to(sendgrid::Destination {
+                address: &email,
+                name: &user.first_name,
+            });
+        match s_g_client.send(reset_password_mail).await {
+            Ok(_) => return Ok("sent email"),
```

Comment:
Should we just return a boolean here? `true` or alternatively add a scalar type to `async-graphql` such as void for methods that should not return anything?

Reply:
boolean is redundant as there is already a separation between error and okay if you want to add a scalar type that is okay too